### ISSUE TITLE
[Publishing] Patch the manifest created by the Tasks.Feed package

### ISSUE THREAD

**Issue by JohnTortugo** [dotnet/arcade#1541]

**Comment by JohnTortugo** [comment]
@ChadNedzlek  - @mmitche told me that you'll probably need to make some changes to the format of the build asset manifest file as part of you work on MissionControl. Is that so? I'll probably make some small changes on the format of the file as well as part of the publishing epic. Do you have any specific change in mind already?

**Comment by ChadNedzlek** [comment]
I think the data we need is there to backform what we want.  I wanted to add the build number that produced the assets there, but there are too many ways to interact with that file, so adding data is basically impossible.  We'll just have to search BAR/Azure DevOps for a build that matches the data we do have (the repository, the commit hash, and the asset version number) and then just assume it's the most recent build where all that is true.

The only real important bit is that we need to sort things into Product/Toolset correctly. In many cases, "Arcade" things are in the "Product" section, when they really belong in the "Toolset" section.

**Comment by ChadNedzlek** [comment]
If "releases" become a thing, we'll need to sync so we figure out how to visualize that on Mission Control, however, which I don't think is accounted for anywhere.  I'll try and sync with you to chat about what's going on once I put out a few fires...

**Comment by JohnTortugo** [comment]
So far I don't think I'll need to change the inner structure of the file, just the root node. Actually, the most important piece for me will be to add the real BuildId (not the label build id) so that I can add the build as a source for the release pipeline. Other thing I'm thinking is that maybe we don't need these other attributes on the <Build> node - not explicitly on the manifest, they can be fetched... So what I am thinking is changing it from this:

`<Build Name="https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-arcade" BuildId="20181212.10" Branch="refs/heads/master" Commit="662be09b3568220ff810a557bbdc8b7d349b4c6a" Location="https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json">`

to this:

`<Build BuildId="https://dnceng.visualstudio.com/internal/_build/results?buildId=57390">`

> I think the data we need is there to backform what we want. I wanted to add the build number that produced the assets there, but there are too many ways to interact with that file, so adding data is prohibitively expensive.

I want to hear more about that. Yeah, let's chat some time.

**Comment by mmitche** [comment]
@ChadNedzlek I don't think you'll need to worry about the release pipelines too much.  They will just update the locations of outputs in the BAR.

Also, the AzDO build id will not be enough, as it won't be unique across instances.  We should just record the full build URI or at the very least account + project + build id.

**Comment by mmitche** [comment]
@JohnTortugo I don't think you need to change the manifest for this.  We'll know the build info (build id and AzDO build link) at the time we do the BAR upload.

**Comment by ChadNedzlek** [comment]
Presumably we'd want to know if the releases failed for some reason (just like we'd want to know if the build failed) and provide links to the relevant releases.  Just having the URI isn't ideal, becaues then we have to write a bunch of parsing code to pull it back out to make any useful calls.  I'd rather store the data in a form we define, so that we aren't guessing at Azure DevOps URL's, since those are prone to changing and causing a huge headache. If we want to store 3 pieces of data, let's store 3 pieces of data.

**Comment by JohnTortugo** [comment]
> @JohnTortugo I don't think you need to change the manifest for this. We'll know the build info (build id and AzDO build link) at the time we do the BAR upload.

Make sense :-)

**Comment by mmitche** [comment]
@ChadNedzlek we can probably cover that with azure devops notifications for now, but keep in mind for future.
